src/core/components/transform.rs

```diff
@@ -55,79 +56,79 @@ impl std::ops::Add for Transform {
     }
 }
 
-#[derive(Clone, Copy, Debug, DeJson)]
-pub struct Scale {
-    pub x: f32,
-    pub y: f32,
-}
+#[derive(Clone, Copy, Debug)]
+pub struct Scale(pub Vec2);
```

Comment:
I think we should keep the x/y fields, they're more readable from a UX perspective